pkg/plugin/plugin.go

```diff
@@ -55,7 +55,7 @@ func (p *plugin) Validate(ctx context.Context, new, old *api.OpenShiftManagedClu
 
 	// if this is an update and not an upgrade, check if we can service it, and
 	// if not, fail early
-	if old != nil && new.Config.PluginVersion != "latest" {
+	if old != nil && new.Properties.ClusterVersion != "latest" {
```

Comment:
I think this makes sense but we need to check it through with MSFT

Reply:
probably needs an extra line of comment here...

Reply:
+1

Reply:
so we have this code in fakerp:

```golang
	switch {
	case cs.Properties.ClusterVersion == "":
	case isAdmin && cs.Properties.ClusterVersion == "latest":
		oldMajor, _, err := parsePluginVersion(cs.Config.PluginVersion)
		if err != nil {
			return nil, err
		}
		if oldMajor < 3 {
			return nil, fmt.Errorf("tried to upgrade a cluster that is too old")
		}
		cs.Properties.ClusterVersion = ""
		cs.Config.PluginVersion = "latest"
	}
```
So we are assuming that MSFT does it in exactly the same order as us
1. Validate()
2. cs.Properties.ClusterVersion = ""
3. cs.Config.PluginVersion = "latest"
4. CreateOrUpdate()